include/z64save.h

```diff
@@ -42,142 +42,193 @@ typedef struct {
-    /* 0x0040 */ u8 magicAcquired;           // "magic_mode"
-    /* 0x0041 */ u8 doubleMagic;             // "magic_ability"
-    /* 0x0042 */ u8 doubleDefense;           // "life_ability"
-    /* 0x0043 */ u8 unk_43;                  // "ocarina_round"
-    /* 0x0044 */ u8 unk_44;                  // "first_memory"
-    /* 0x0046 */ u16 owlActivationFlags;     // "memory_warp_point"
-    /* 0x0048 */ u8 unk_48;                  // "last_warp_pt"
-    /* 0x004A */ s16 savedSceneNum;          // "scene_data_ID"
+    /* 0x0024 */ SavePlayerData playerData;
     /* 0x004C */ ItemEquips equips;
     /* 0x0070 */ Inventory inventory;
-    /* 0x00F8 */ u32 roomInf[128][7];
-    /* 0x0EF8 */ u8 weekEventReg[100];       // "week_event_reg"
-    /* 0x0F5C */ u32 mapsVisited;            // "area_arrival"
-    /* 0x0F60 */ u32 unk_F60;                // "cloud_clear"
-    /* 0x0F64 */ u8 unk_F64;                 // "oca_rec_flag"
-    /* 0x0F65 */ u8 unk_F65;                 // "oca_rec_flag8"
-    /* 0x0F66 */ u8 unk_F66[128];            // "oca_rec_buff8"
-    /* 0x0FE6 */ s8 unk_FE6;                 // "aikotoba_index"
-    /* 0x0FE7 */ s8 unk_FE7[5];              // "aikotoba_table"
-    /* 0x0FEC */ s8 lotteryCodes[3][3];      // "numbers_table"
-    /* 0x0FF5 */ s8 spiderHouseMaskOrder[6]; // "kinsta_color_table"
-    /* 0x0FFB */ s8 bomberCode[5];           // "bombers_aikotoba_table"
+    /* 0x00F8 */ PermanentSceneFlags permanentSceneFlags[120];
+    /* 0x0E18 */ u8 unk_E18[0x54];
+    /* 0x0E6C */ u32 dekuPlaygroundHighScores[3];
+    /* 0x0E78 */ u32 pictoFlags0;
+    /* 0x0E7C */ u32 pictoFlags1;
+    /* 0x0E80 */ u32 unk_E80;
+    /* 0x0E84 */ u32 unk_E84;
+    /* 0x0E88 */ u32 unk_E88[7]; // Invadepoh flags
+    /* 0x0EA4 */ u32 scenesVisible[7]; // tingle maps and clouded regions on pause map. Stores scenes bitwise for up to 224 scenes even though there are not that many scenes
+    /* 0x0EC0 */ u32 skullTokenCount; // upper 16 bits store Swamp skulls, lower 16 bits store Ocean skulls
+    /* 0x0EC4 */ u32 unk_EC4;
+    /* 0x0EC8 */ u32 unk_EC8;
+    /* 0x0ECC */ u32 unk_ECC[2];
+    /* 0x0ED4 */ u32 stolenItems; // Items stolen by Takkuri and given to Curiosity Shop Man
+    /* 0x0ED8 */ u32 unk_DD8;
+    /* 0x0EDC */ u32 bankRupees;
+    /* 0x0EE0 */ u32 unk_EE0;
+    /* 0x0EE4 */ u32 unk_EE4;
+    /* 0x0EE8 */ u32 unk_EE8;
+    /* 0x0EEC */ u32 horseBackBalloonHighScore;
+    /* 0x0EF0 */ u32 unk_EF0;
+    /* 0x0EF4 */ u32 unk_EF4; // Shooting Gallery Man Flags
+    /* 0x0EF8 */ u8 weekEventReg[100];           // "week_event_reg"
+    /* 0x0F5C */ u32 mapsVisited;                // "area_arrival"
+    /* 0x0F60 */ u32 mapsVisible;                // "cloud_clear"
+    /* 0x0F64 */ u8 unk_F64;                     // "oca_rec_flag                   has scarecrows song
+    /* 0x0F65 */ u8 unk_F65;                     // "oca_rec_flag8                  scarecrows song set?
+    /* 0x0F66 */ u8 scarecrowsSong[128];         // "oca_rec_buff8"
```

Comment:
One thing I'd like to add about just calling this `scarecrowsSong` is that while MM only practically using 1 scarecrow song instead of OoT's short and long version, Both versions of scarecrow's songs are actually in MM too, it's just that the long one remains unused in practice (there's some interesting unused code that I'm working on deciphering though). That said, since both are still present in the game, I still think it's worth choosing a name that can distinguish the two since there will be variables needed for both. Do you know if this is the used short one or the unused long one?

Reply:
No, I don't know enough of the scarecrows song to know which one is this one